src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -4227,6 +4247,43 @@ GenTree* Lowering::LowerHWIntrinsicCndSel(GenTreeHWIntrinsic* cndSelNode)
     return cndSelNode->gtNext;
 }
 
+GenTree* Lowering::LowerHWIntrinsicConvertVectorToMask(GenTreeHWIntrinsic* mask)
+{
+    assert(mask->OperIsHWIntrinsic(NI_Sve_ConvertVectorToMask));
+
+    GenTree* op1 = mask->Op(1);
+    GenTree* op2 = mask->Op(2);
+
+    if (op2->IsVectorZero())
```

Comment:
This feels unnecessary. This is a constant folding we should have handled earlier and had produce a `CNS_MSK` instead.

We should then be handling the `CNS_MSK` in codegen and specializing it for Zero and AllBitsSet to generate the instructions directly.

Lowering it to be an intrinsic is strictly worse and won't participate in the LSRA optimizations for constant values

Reply:
> We should then be handling the `CNS_MSK` in codegen and specializing it for Zero and AllBitsSet to generate the instructions directly.

Ok, so in lowering we would contain the `CNS_MSK` inside the Convert. Then in codegen of the convert it'd generate the single instruction.

Reply:
I would expect the JIT to have constant folded this to be a CNS_MSK much earlier, such as in morph or import.

We have a number of trivial constant foldings we do even in MinOpts

Reply:
> Ah, right. I forgot that Arm64 has a bit of a different layout and checks every nth bit instead of compressing them to be sequential.
> 
> We could have CNS_MSK track a base type if that was beneficial, but not something that we should block this PR on.

There's the additional issue that `createTrueMaskInt32(ALL)` isn't all 1s, but instead 1000,1000, etc. So regardless, if we're going to import true masks as const vectors, then it's going to need some parsing to reconstruct back to a mask in codegen. Latest version of the PR does that.

Now I've got the issue that some of the optimisations in lowering never trigger because they are looking for a vector which is `isAllBitsSet()`. That can be fixed up.